extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -191,168 +238,54 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
-					let pathToMarkdown = path.join(root, 'content', sections[i].url.concat('.md'));
-					// Note: Currently, if there is an ipynb and a md file with the same name, Jupyter Books only shows the notebook.
-					// Following Jupyter Books behavior for now
-					if (fs.existsSync(pathToNotebook)) {
-						let notebook = new BookTreeItem({
-							title: sections[i].title,
-							root: root,
-							tableOfContents: tableOfContents,
-							page: sections[i],
-							type: BookTreeItemType.Notebook,
-							treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed
-						},
-							{
-								light: this._extensionContext.asAbsolutePath('resources/light/notebook.svg'),
-								dark: this._extensionContext.asAbsolutePath('resources/dark/notebook_inverse.svg')
-							}
-						);
-						notebooks.push(notebook);
-						this._allNotebooks.set(pathToNotebook, notebook);
-						if (this._openAsUntitled) {
-							this._allNotebooks.set(path.basename(pathToNotebook), notebook);
-						}
-					} else if (fs.existsSync(pathToMarkdown)) {
-						let markdown = new BookTreeItem({
-							title: sections[i].title,
-							root: root,
-							tableOfContents: tableOfContents,
-							page: sections[i],
-							type: BookTreeItemType.Markdown,
-							treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed
-						},
-							{
-								light: this._extensionContext.asAbsolutePath('resources/light/markdown.svg'),
-								dark: this._extensionContext.asAbsolutePath('resources/dark/markdown_inverse.svg')
-							}
-						);
-						notebooks.push(markdown);
-					} else {
-						let error = localize('missingFileError', 'Missing file : {0}', sections[i].title);
-						this._errorMessage = error;
-						vscode.window.showErrorMessage(error);
-					}
-				}
-			} else {
-				// TODO: search functionality (#6160)
-			}
+		else if (element.root.endsWith('.ipynb')) {
+			let baseName: string = path.basename(element.root);
+			parentPath = element.root.replace(baseName, 'readme.md');
 		}
-		return notebooks;
-	}
-
-	getNavigation(uri: vscode.Uri): Thenable<azdata.nb.NavigationResult> {
-		let notebook = uri.scheme !== 'untitled' ? this._allNotebooks.get(uri.fsPath) : this._allNotebooks.get(path.basename(uri.fsPath));
-		let result: azdata.nb.NavigationResult;
-		if (notebook) {
-			result = {
-				hasNavigation: true,
-				previous: notebook.previousUri ? this._openAsUntitled ? vscode.Uri.parse(notebook.previousUri).with({ scheme: 'untitled' }) : vscode.Uri.file(notebook.previousUri) : undefined,
-				next: notebook.nextUri ? this._openAsUntitled ? vscode.Uri.parse(notebook.nextUri).with({ scheme: 'untitled' }) : vscode.Uri.file(notebook.nextUri) : undefined
-			};
-		} else {
-			result = {
-				hasNavigation: false,
-				previous: undefined,
-				next: undefined
-			};
+		else {
+			return undefined;
 		}
-		return Promise.resolve(result);
+		return Promise.resolve(this.currentBook.getAllBooks().get(parentPath));
 	}
```

Comment:
Any particular reason you're mixing returning Promises and undefined here? You should really just be sticking to one. Since this doesn't seem to require being async you should be able to just return the values directly